### ISSUE TITLE
PubNub element update proposal

### ISSUE THREAD

**Issue by iiegor** [pubnub/pubnub-polymer#7]
## PubNub element update proposal

I know already exists something similar (see #5) but this is a better and proper update to Polymer 1.1
### Actual issues
- Use of a deprecated polyfill (polymer/platform) we need to use webcomponentsjs instead.
- The code style is messed up.
- Core elements were for the old versions of polymer, let's use a better name.
- Actually, the name is very confussing... pubnub-polymer, core-pubnub, pubnub-element. 
- All the sub-elements are in the same file, separating them in the corresponding files it's better for developers.
- Need to set a version for the Polymer dependency.
- Restructure the project files / Better organization of the files.
- Change the pubnub element data tags to a proper html structure, ex: publish_key => publish-key.
- Documentation errors and code typos.
- Vulcanization problems
- Tests are missing!
- subscribeChanged not triggering using Polymer 1.0 and the latest version of the pubnub element (see https://github.com/pubnub/pubnub-polymer/issues/7#issuecomment-146344105)
### Result

The functionality is the same and the use is almost similar:

``` html
<pubnub-element 
   publish-key="pub-X-XXXXX-XXXX-XXX-XXXX-XXXXXXXX" 
   subscribe-key="sub-X-XXXXX-XXXX-XXX-XXXX-XXXXXXXX">

  <pubnub-subscribe 
    channel="polymer-dev" 
    messages="{{messages}}" 
    on-callback="_subscribeCallback"></pubnub-subscribe>
</pubnub-element>
```

Due this is a breaking/important change I think this must be bumped as 4.1.0 (Major.Minor.Patch).

**Comment by Whyser** [comment]
And to add to the list (correct me if I'm wrong but the following event won't trigger in Polymer 1.0):

subscribeChanged: function(oldValue, newValue) {
        if (newValue) {
          this.sub();
        }
        else {
          this.pubnub.unsubscribe({
            channel: this.channel  
          });
        }
      }

The variable subcribe should have an observer declared to invoke this method (or some name of it). At the moment it does nothing as far as I can tell.

Sorry if this should be it's own issue.

**Comment by iiegor** [comment]
@Whyser Yes, you are right. The author of the code didn't fix that when upgrade from Polymer 0.5. I will add that to the list, thanks!

**Comment by girliemac** [comment]
Thanks, I am looking into the PR (#8) now.
and I agree that the naming convention is pretty bad. I always wanted to fix it.

I keep this issue opened for a while, until everything is resolved.
